services/organization.js

```diff
@@ -50,9 +84,106 @@ class Organization {
    * @param {Filters} [filters] Filters to apply to the query
    * @returns {Promise<Object[]>} An array of Organizations
    */
-  async listOrganizations(filters = []) {
+  async listOrganizations(filters = {}) {
     return await this.organizationCollection.aggregate([{ "$match": filters }]);
   }
+
+  /**
+   * Edit Organization API Interface.
+   *
+   * - `ADMIN` can call this API only
+   *
+   * @api
+   * @param {EditOrganizationInput} params Endpoint parameters
+   * @param {{ cookie: Object, userInfo: Object }} context API request context
+   * @returns {Promise<Object>} The modified organization
+   */
+  async editOrganizationAPI(params, context) {
+    if (!context?.userInfo?.email || !context?.userInfo?.IDP) {
+      throw new Error(ERROR.NOT_LOGGED_IN);
+    }
+    if (context?.userInfo?.role !== USER.ROLES.ADMIN) {
+        throw new Error(ERROR.INVALID_ROLE);
+    }
+    if (!params?.orgID) {
+        throw new Error(ERROR.INVALID_ORG_ID);
+    }
+
+    return this.editOrganization(params.orgID, params);
+  }
+
+  /**
+   * Edit an organization by it's `_id` and a set of parameters
+   *
+   * @async
+   * @typedef {{ orgID: string, name: string, conciergeID: string, studies: Object[], status: string }} EditOrganizationInput
+   * @throws {Error} If the organization is not found or the update fails
+   * @param {string} orgID The ID of the organization to edit
+   * @param {EditOrganizationInput} params The organization input
+   * @returns {Promise<Object>} The modified organization
+   */
+  async editOrganization(orgID, params) {
+      const currentOrg = await this.getOrganizationByID(orgID);
+      const updatedOrg = { updateAt: getCurrentTime() };
+      if (!currentOrg) {
+          throw new Error(ERROR.ORG_NOT_FOUND);
+      }
+
+      if (params.name && params.name !== currentOrg.name) {
+          const existingOrg = await this.getOrganizationByName(params.name);
+          if (existingOrg) {
+              throw new Error(ERROR.DUPLICATE_ORG_NAME);
+          }
+          updatedOrg.name = params.name;
+      }
+
+      if (params.conciergeID && params.conciergeID !== currentOrg.conciergeID) {
```

Comment:
@n2iw @AustinSMueller Can you confirm if we need to update the user collection when assigning/unassigning a curator from an organization?

The `User` definition has a `curatedOrganizations` field. Is this used somewhere?

https://github.com/CBIIT/crdc-datahub-authz/blob/2.0.0/resources/graphql/authorization.graphql#L11-L13

Reply:
Confirmed it's not necessary.